benchmark/napi/function_args/binding.cc

```diff
@@ -0,0 +1,100 @@
+#include <v8.h>
+#include <node.h>
+#include <assert.h>
+
+using v8::Local;
+using v8::Value;
+using v8::Number;
+using v8::String;
+using v8::Object;
+using v8::Array;
+using v8::ArrayBufferView;
+using v8::ArrayBuffer;
+using v8::FunctionCallbackInfo;
+
+void CallWithString(const FunctionCallbackInfo<Value>& args) {
+  assert(args.Length() == 1 && args[0]->IsString());
+  if (args.Length() == 1 && args[0]->IsString()) {
+    Local<String> str = args[0].As<String>();
+    const int32_t length = str->Utf8Length() + 1;
+    char* buf = new char[length];
+    str->WriteUtf8(buf, length);
+    delete [] buf;
+  }
+}
+
+void CallWithArray(const FunctionCallbackInfo<Value>& args) {
+  assert(args.Length() == 1 && args[0]->IsArray());
+  if (args.Length() == 1 && args[0]->IsArray()) {
+    const Local<Array> array = args[0].As<Array>();
+    uint32_t length = array->Length();
+    for (uint32_t i = 0; i < length; ++ i) {
+      Local<Value> v = array->Get(i);
+    }
+  }
+}
+
+void CallWithNumber(const FunctionCallbackInfo<Value>& args) {
+  assert(args.Length() == 1 && args[0]->IsNumber());
+  if (args.Length() == 1 && args[0]->IsNumber()) {
+    double value = args[0].As<Number>()->Value();
+  }
+}
+
+void CallWithObject(const FunctionCallbackInfo<Value>& args) {
+  v8::Isolate* isolate = args.GetIsolate();
+
+  assert(args.Length() == 1 && args[0]->IsObject());
+  if (args.Length() == 1 && args[0]->IsObject()) {
```

Comment:
I think it might be best to save the result of `args.Length() == 1 && args[0]->IsObject()` first and then both `assert()` it and use it as the if-statement condition. For clarity, you could `assert(argumentsCorrect && "argument length is 1 and the first argument is an object")`, where `bool argumentsCorrect = (args.Length() == 1 && args[0]->IsObject());`